src/Admin/Vue.php

```diff
@@ -27,26 +33,24 @@ public function init() {
 			$logged = $this->payplug_section_logged();
 			$payplug = (new PayplugGateway());
 
-			$payplug_wooc_settings = get_option( 'woocommerce_payplug_settings', [] );
-
-			if ((empty($payplug_wooc_settings['oney_thresholds_default_min'])) && (empty($payplug_wooc_settings['oney_thresholds_default_max']))) {
-				$payplug_wooc_settings['oney_thresholds_default_min'] = $payplug->min_oney_price;
-				$payplug_wooc_settings['oney_thresholds_default_max'] = $payplug->max_oney_price;
+			if ((empty($this->options['oney_thresholds_default_min'])) && (empty($this->options['oney_thresholds_default_max']))) {
```

Comment:
Wondering if we can use the `$payplug->settings`, it has all the options we're using, it should be more performant am I right?

Reply:
this will need a new PayplugGateway everytime we are calling the method

Reply:
you already have it on the line 28 i beleive

`$payplug = (new PayplugGateway());`